client_code/session.py

```diff
@@ -14,6 +16,25 @@ def tabulator_options():
     }
 
 
+def sync_storage(scroll_store, local_store):
+    scroll_ballots = scroll_store.all()
+    local_ballots = local_store.all(Ballot)
+    for ballot in scroll_ballots:
+        if ballot.uuid not in [b.uuid for b in local_ballots]:
+            local_store.save(ballot)
+
+
+def sync_ballot(scroll_store, local_store, ballot_id):
+    local_ballots = local_store.all(Ballot)
+    if ballot_id not in [b.uuid for b in local_ballots]:
+        try:
+            ballot = scroll_store.get_ballot(ballot_id)
+            local_store.save(ballot)
```

Comment:
Side effects within functions are best avoided where possible. This function should probably just return the Ballot instance and leave the caching to the caller.